lib/App.js

```diff
@@ -176,34 +176,74 @@ App.prototype.loadViews = function() {};
 
 App.prototype.loadStyles = function() {};
 
+// This function is defined by requiring 'derby/parsing'
+App.prototype.addViews = null;
+
 App.prototype.component = function(viewName, constructor) {
   if (typeof viewName === 'function') {
     constructor = viewName;
     viewName = null;
   }
-  // DEPRECATED: constructor.prototype.view and constructor.prototype.name
-  var viewFilename = constructor.view || constructor.prototype.view;
-  var constructorIs = constructor.is || constructor.prototype.name;
+
+  var viewArg = constructor.view;
+  var viewIs, viewFilename, viewSource, viewDependencies;
+  // Always using an object for the static `view` property is preferred
+  if (viewArg && typeof viewArg === 'object') {
+    viewIs = viewArg.is;
+    viewFilename = viewArg.file;
+    viewSource = viewArg.source;
+    viewDependencies = viewArg.dependencies;
+  } else {
+    // Ignore other properties when `view` is an object. It is possible that
+    // properties could be inherited from a parent component when extending it.
+    //
+    // DEPRECATED: constructor.prototype.name and constructor.prototype.view
+    // use the equivalent static properties instead
+    viewIs = constructor.is || constructor.prototype.name;
+    viewFilename = constructor.view || constructor.prototype.view;
+  }
+
+  if (viewFilename && viewSource) {
+    throw new Error('Component may not specify both a view file and source');
+  }
+
+  if (viewDependencies) {
+    for (var i = 0; i < viewDependencies.length; i++) {
+      var dependency = viewDependencies[i];
+      if (Array.isArray(dependency)) {
+        this.component(dependency[0], dependency[1]);
```

Comment:
`App#component` on the server can do some synchronous filesystem operations, so it would be nice to avoid them if a component is already registered with the same view name. Especially for cases where a set of base re-usable components are referenced by many other components.

Changing the behavior of the main `App#component` call would technically be a breaking change, as today it's last call wins for a given view name, but we could have `dependencies` not overwrite as it's a new feature.

Reply:
👍